src/Posts.js

```diff
@@ -1,31 +1,33 @@
-import React from 'react'
-import axios from 'axios'
+import React, { useState, useEffect } from "react";
 
-export default class Posts extends React.Component {
-    state = {
-        posts: []
-    }
+export default function Posts(props) {
+    const [posts, setPosts] = useState([
+        {
+            posts: []
+        }
+    ]);
 
-    componentDidMount() {
-        axios.get('http://0.0.0.0:3000/posts.json')
-            .then(res => {
-                const posts = res.data;
-                this.setState({ posts });
-            })
+    async function fetchPostsData() {
+        const response = await fetch(
+            `http://0.0.0.0:3000/posts.json`
+        );
+        setPosts(await response.json());
     }
 
-    render() {
-        return (
-            <ul>
-                { this.state.posts.map(post =>
-                    <div key={post.id}>
-                        <div>
-                            <h3>{post.title}</h3>
-                            <div dangerouslySetInnerHTML={{ __html: post.rich_body }}></div>
-                        </div>
+    useEffect(() => {
+        fetchPostsData(props.id);
+    }, [props.id]);
+
+    return (
+        <ul>
+            { posts.map(post =>
+                <div key={post.id}>
+                    <div>
+                        <h3>{post.title}</h3>
+                        <div dangerouslySetInnerHTML={{__html: post.rich_body}}/>
```

Comment:
dangerously? Because it's not sanitised?

Reply:
Correct.

Reply:
Yeah, I would look into sanitising it, or being maybe breaking it down some other way to be injected

Reply:
I'll make a task to do so. This isn't being merged into develop yet.